packages/dev/core/src/Meshes/Builders/greasedLineBuilder.ts

```diff
@@ -197,7 +197,26 @@ export function CreateGreasedLine(name: string, options: GreasedLineMeshBuilderO
         }
     }
 
-    if (!options.instance) {
+    if (options.instance) {
+        // update the data on the mesh instance
+        instance = options.instance;
+        if (instance instanceof GreasedLineRibbonMesh) {
+            instance.addPoints(allPoints, initialGreasedLineOptions);
+        } else {
+            const currentWidths = instance.widths;
+
+            if (currentWidths) {
+                const newWidths = currentWidths.slice();
+                for (const w of widths) {
+                    newWidths.push(w);
+                }
+                instance.widths = newWidths;
+            } else {
+                instance.widths = widths;
+            }
+            instance.addPoints(allPoints);
+        }
+    } else {

```

Comment:
What's the point of swapping the two branches?

Reply:
probably saves a ! ... but pretty sure it won t change perf